e2e/_support/fixtures.ts

```diff
@@ -0,0 +1,67 @@
+import { test as base, Page } from "@playwright/test";
+import { User, userPage } from "./auth";
+import { FactoryBot } from "./factorybot";
+
+class UserFixture {
+  page: Page;
+  user: User;
+
+  constructor(page: Page, user: User) {
+    this.page = page;
+    this.user = user;
+  }
+}
+
+type MaMpfFixtures = {
+  student: UserFixture;
+  student2: UserFixture;
+  admin: UserFixture;
+  teacher: UserFixture;
+  tutor: UserFixture;
+  factory: FactoryBot;
+};
+
+export * from "@playwright/test";
+export const test = base.extend<MaMpfFixtures>({
+  student: async ({ browser }, use) => {
+    const [user, page, browserContext] = await userPage(browser, "student");
+    const fixture = new UserFixture(page, user);
+    await use(fixture);
+    await browserContext.close();
+  },
```

Comment:
Learn more about Playwright fixtures [here](https://playwright.dev/docs/test-fixtures#overriding-fixtures). Coming from Cypress, this is so much nicer to have than a general `commands.js` file.